### ISSUE TITLE
Decide how to pass on "count"

### ISSUE THREAD

**Issue by mhvk** [mhvk/scintillometry#48]
We briefly had a `BaseData` subclass of `ndarray` with a `count` attribute (#44), but this was rather problematic, so for now we are using a structured dtype with 'data' and 'count' entries (#38). Should think if that is really what we want.

**Comment by mhvk** [comment]
While thinking about what to do with `count` in, e.g., dedispersion: possibly, it should just be a float so that it can be dedispersed. Indeed, it might make sense to more generally call it "weight".

**Comment by luojing1211** [comment]
If we are going to use "The weight" as a data mask. Can we make something like a masked array?

**Comment by mhvk** [comment]
> Can we make something like a masked array?

That would be a good solution. Ideally we implement something that can replace the data we have and "just work" for tasks like `Dedispersion`. Alternatively, we could add a 2-item dimension to the sample shape.

**Comment by luojing1211** [comment]
I am trying to picture how does out pipeline handle the mask/weight. The weight array flows with the pipeline. Another interesting question is should we include the weight to our output file, psrfits file.

**Comment by mhvk** [comment]
Yes, my hope was that by having a float weight, one could just "dedisperse" it, etc. Ideally, we do indeed have the weight in the output file too.
